### ISSUE TITLE
Documents must have <title> element to aid in navigation Fails on title in head tag

### ISSUE THREAD

**Issue by Rathch** [dequelabs/axe-core#4580]
### Product

axe-core

### Latest Version

- [X] I have tested the issue with the latest version of the product

### Issue Description

The test fails on not recognizing the title tag when it's not in HTML root but a child of the head tag
```
<!doctype html>
<html lang="en">
  <head>
    <title>This will fail</title>     
  </head>
  <body>
      ...   
  </body> 
</html>
```

```
<!doctype html>
<html lang="en">
  <title>This will work</title>
  <head>
         
  </head>
  <body>
      ...   
  </body> 
</html>
```
but it's valid and best practice to have it like this https://www.w3.org/WAI/WCAG21/Techniques/html/H25.html

**Comment by straker** [comment]
Thanks for the issue. The reason for this is that browsers automatically reformat the invalid HTML into valid HTML by placing the `title` into the `head` element. So this:

```html
<!doctype html>
<html lang="en">
  <title>This will work</title>
  <head>
         
  </head>
  <body>
      ...   
  </body> 
</html>
```

Is changed into the following when rendered:

```html
<html lang="en">
  <head>
    <title>This will work</title>
  </head>
  <body>
    <text>
      ...
    </text>
  </body>
</html>
```

**Comment by straker** [comment]
I'm afraid I don't quite understand. So the rule that checks for title is the `document-title` rule. Are you saying that that rule is failing to find a `title` element even when the title is in the head? 

I'm not sure how the test would fail as we use `document.title` to determine if there is a title. Would it be possible to see the page where the issue is occurring? Or possibly a minimal setup example with Playwright-axe that shows the issue? When I open both HTML examples provided in the original issue comment, axe-core passes `document-title`.

**Comment by Rathch** [comment]
https://github.com/Rathch/playwright-base-tests-cucumber/tree/only-title-test

I isolated the issue in this branch

**Comment by Rathch** [comment]
for me it happend on the startpage i isolated the specific test in the branch only-title-test
